api/agent/drivers/docker/cookie.go

```diff
@@ -140,6 +140,56 @@ func (c *cookie) configurePIDs(log logrus.FieldLogger) {
 	c.opts.HostConfig.PidsLimit = &pids64
 }
 
+// configureOpenFiles will set the ULimit for `nofile` on the Docker container
+func (c *cookie) configureOpenFiles(log logrus.FieldLogger) {
```

Comment:
Potential idea for reducing the 4 all very similar functions.  What do you think about something like the following?

```
type ulimitSetter func()(*uint64, string)

func (c *cookie) configureUlimit(usf ulimitSetter, log logrus.FieldLogger) {
    ulimitValP, ulimitName := usf()
    if ulimitValP == nil {
        return
    }

    ulimit64 := int64(*ulimitValP)
    // bail out if ulimit is greater that MaxInt64
    if ulimit64 < 0 {
        return
    }

    log.WithFields(logrus.Fields{ulimitName: ulimit64, "call_id": c.task.Id()}).Debugf("setting %s", ulimitName)
    c.addULimit(docker.ULimit{Name: ulimitName, Soft: ulimit64, Hard: ulimit64})
} 
```

This does make the logging slightly more gross, but it means you can do a common set with 1/4th as much code.

If you decided to go this route, you'd also need to change the interfaces in `driver.go`, tweak the functions in `agent.go` to return the ulimit's name in addition to the value, and then  modify the config bits in `docker.go` to look something like:
```
cookie.configureUlimit(task.MessageQueue(), log)
```
As one example.

Reply:
Didn't go with the setterFunc route but just a single method that gets called multiple times. Let me know what you think of that solution.

Reply:
https://github.com/fnproject/fn/pull/1495/commits/45dbdc66351d56af4629f738529dbf7384550336

Reply:
Thanks; like the change here.  I realized after I posted my comments that the setterFunc was just needless complication, so I'm glad you didn't do that part.